### PR TITLE
chore: merge in strict ntsc checks

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- migrate 2 rbac interfaces defined under `UserAuthZ` to `NSCAuthZ`
- ntsc proxy and events to behave as before: 8751 & 8733

works toward [DET-8732]

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-8732]: https://determinedai.atlassian.net/browse/DET-8732?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/command/events.go [hunk 1]
```diff
@@ -135,15 +134,21 @@ func (e *eventManager) Receive(ctx *actor.Context) error {
 func canAccessCommandEvents(ctx *actor.Context, c echo.Context) error {
 	curUser := c.(*context.DetContext).MustGetUser()
 	taskID := model.TaskID(ctx.Self().Parent().Address().Local())
-	ownerID, err := db.GetCommandOwnerID(c.Request().Context(), taskID)
```
Comment (hamidzr) [inline master/internal/command/events.go:138]: this was assuming that the snapshot is always available. I'll check when we do this

File: master/internal/api_experiment.go [hunk 2]
```diff
@@ -36,7 +36,7 @@ import (
 	"github.com/determined-ai/determined/master/internal/hpimportance"
 	"github.com/determined-ai/determined/master/internal/lttb"
 	"github.com/determined-ai/determined/master/pkg/actor"
-	command "github.com/determined-ai/determined/master/pkg/command"
+	pkgCommand "github.com/determined-ai/determined/master/pkg/command"
```
Comment (hamidzr) [inline master/internal/api_experiment.go:39]: not sure if we want this one as well.

**Review by eecsliu — APPROVED** [review]
LGTM! Note: reminder to check on command snapshot

File: master/internal/command/events.go (hunk 1 above)
Comment (hamidzr) [reply master/internal/command/events.go:138]: commands are persisted before they come online. we should be good to rely on this
https://github.com/hamidzr/determined/blob/040be76a417f4a1a5ff6626e15d199341a8a9540/master/internal/command/command.go#L328

File: master/Makefile [hunk 3]
```diff
@@ -35,7 +35,9 @@ build/mock_gen.stamp: $(MOCK_GENERATION_INPUTS)
 	mockery --name=UserAuthZ --dir=internal/user --output internal/mocks --filename user_authz_iface.go
 	mockery --name=WorkspaceAuthZ --dir=internal/workspace --output internal/mocks --filename workspace_authz_iface.go
 	mockery --name=ProjectAuthZ --dir=internal/project --output internal/mocks --filename project_authz_iface.go
+	# CHECK: which naming convention do we want? cc: @nick
```
Comment (hamidzr) [inline master/Makefile:38]: cc @NicholasBlaskey (for later, as this is going into a feature branch)

**Pull request merged**